x-pack/plugins/infra/public/pages/metrics/inventory_view/components/waffle/conditional_tooltip.tsx

```diff
@@ -29,10 +28,13 @@ export interface Props {
   currentTime: number;
   node: InfraWaffleMapNode;
   nodeType: InventoryItemType;
-  theme: EuiTheme | undefined;
 }
-export const ConditionalToolTip = withTheme(({ theme, node, nodeType, currentTime }: Props) => {
+
+export const ConditionalToolTip = ({ node, nodeType, currentTime }: Props) => {
+  const { euiTheme } = useEuiTheme();
   const { sourceId } = useSourceContext();
+  // prevents auto-refresh from cancelling ongoing requests to fetch the data for the tooltip
+  const [requestTs] = useState(currentTime);
```

Comment:
Every time the auto-refresh kicks in, a new `currentTime` will be passed into the `conditional_tooltip`. 

If the tooltip is loading and receives a new `currentTime`, it will make the `useSnapshot` hook to make a new request. This new request will cause the previous one to be aborted. We want the `conditional_tooltip` to use the first known `currentTime` it receives for the entirety of its lifecycle so that it can complete the request successfully.

This is what what happen without this:
https://github.com/elastic/kibana/assets/2767137/bea5a87f-565d-47af-a827-2546969dba88






https://github.com/elastic/kibana/assets/2767137/25b6f226-f469-4959-a3ee-a80f2369b46b

Reply:
Got it thanks for explaining and adding a ref there. I checked it again and also saw that the values in the tooltip can be different depending on when you open it - after stopping the auto-refresh and opening the tooltip several seconds later the numbers in the tooltip will be different compared to the waffle values (it's off by 0,2-0,5 but sometimes it's more):
![image](https://github.com/elastic/kibana/assets/14139027/de20e6da-474d-4bb6-9f09-01016e0d349c)

I am not sure if that's related to your changes, it could be that it was like that. In the recording you can see the steps better: 

https://github.com/elastic/kibana/assets/14139027/81dae21a-41a5-4be5-926c-ba703ace3227

Reply:
I reproduced it on edge - you can ignore that as it's not related to the PR - maybe we can still have a new issue for that. I am not sure what is the correct solution - maybe setting always the same time in both the tooltip and the waffle and checking if stopping the auto-refresh is not affecting the tooltip 🤔